File: src/main/java/com/example/fileservice/dto/FileUploadRequest.java
package com.example.fileservice.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.web.multipart.MultipartFile;

import java.util.Date;

@Getter
@Setter
public class FileUploadRequest {

    @NotBlank(message = "Name cannot be blank")
    private String name;

    @NotBlank(message = "Content type cannot be blank")
    private String contentType;

    @NotBlank(message = "Meta cannot be blank")
    private String meta;

    @NotBlank(message = "Source cannot be blank")
    private String source;

    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
    private Date expireTime;

    @NotNull(message = "File content cannot be null")
    private MultipartFile content;
}

File: src/main/kotlin/com/example/fileservice/data/dto/FileUploadRequest.kt
package com.example.fileservice.data.dto

import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull
import org.springframework.format.annotation.DateTimeFormat
import org.springframework.web.multipart.MultipartFile

data class FileUploadRequest(

    @field:NotBlank(message = "Name cannot be blank") val name: String,

    @field:NotBlank(message = "Content type cannot be blank") val contentType: String,

    @field:NotBlank(message = "Meta cannot be blank") val meta: String,

    @field:NotBlank(message = "Source cannot be blank") val source: String,

    @field:DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) val expireTime: String?,

    @field:NotNull(message = "File content cannot be null") val content: MultipartFile?
)
